### ISSUE TITLE
Site editor - Add keyboard shortcut for saving

### ISSUE THREAD

**Issue by talldan** [WordPress/gutenberg#36020]
## What problem does this address?
I keep pressing Cmd+S in the site editor and seeing this unhelpful browser dialog:
<img width="343" alt="Screenshot 2021-10-28 at 11 55 43 am" src="https://user-images.githubusercontent.com/677833/139184112-9f607323-a6ef-44b1-bee7-314ad997088e.png">

I think this will be quite frustrating for other users too.

## What is your proposed solution?
Add a keyboard shortcut for saving.

I realise saving in the site editor is a little more complicated, but I think it could at least open the entity saving sidebar thing and maybe focus the save button.

The other thing I notice is there's a lack of snackbar notification when saving, and so there's probably also a missing screenreader spoken message.

**Comment by annezazu** [comment]
Closing this out as a duplicate: https://github.com/WordPress/gutenberg/issues/29642 Will add the other to must haves.
